azure-ipam/ipam.go

```diff
@@ -85,17 +106,25 @@ func (p *IPAMPlugin) CmdAdd(args *cniSkel.CmdArgs) error {
 		p.logger.Error("Failed to interpret CNS IPConfigResponse", zap.Error(err), zap.Any("response", resp))
 		return cniTypes.NewError(ErrProcessIPConfigResponse, err.Error(), "failed to interpret CNS IPConfigResponse")
 	}
-	p.logger.Debug("Parsed pod IP", zap.String("podIPNet", podIPNet.String()))
-
-	cniResult := &types100.Result{
-		IPs: []*types100.IPConfig{
-			{
-				Address: net.IPNet{
-					IP:   net.ParseIP(podIPNet.Addr().String()),
-					Mask: net.CIDRMask(podIPNet.Bits(), 32), // nolint
-				},
-			},
-		},
+	for _, IPNet := range *podIPNet {
+		p.logger.Debug("Parsed pod IP", zap.String("podIPNet", IPNet.String()))
```

Comment:
this should be in the loop starting on L114 and this loop should be removed - no need to iterate over the same thing twice esp. for debug logs only

Reply:
Removed the loop and moved the statement to the start of the following loop.